validation/src/main/scala/hmda/validation/ValidationStats.scala

```diff
@@ -22,7 +23,14 @@ object ValidationStats {
   case class ValidationStatsState(stats: Seq[SubmissionStats] = Nil) {
     def updated(event: Event): ValidationStatsState = event match {
       case SubmissionStatsAdded(s) =>
-        ValidationStatsState(stats :+ s)
+        val matchingSubs = stats.filter(stat => stat.id == s.id)
+        if (matchingSubs.isEmpty) {
+          ValidationStatsState(stats :+ s)
+        } else {
+          val oldSub = matchingSubs.head
+          val newSub = oldSub.copy(id = s.id, totalLars = oldSub.totalLars + s.totalLars, taxId = oldSub.taxId + s.taxId)
```

Comment:
Okay, I see what it's doing. It seems like this `SubmissionStatsAdded` event is doing two different jobs. `totalLars` and `taxId` will never get added at the same time, so we don't need this method to do both. I think if we had something like `SubmissionTotalsAdded` and `SubmissionTaxIdAdded` separately, the purpose of the code would be much clearer.